web_poet/page_inputs/http.py

```diff
@@ -18,13 +19,64 @@
 _AnyStrDict = Dict[AnyStr, Union[AnyStr, List[AnyStr], Tuple[AnyStr, ...]]]
 
 
-class ResponseURL(str):
-    """ URL of the response """
+class _Url:
+    def __init__(self, url: Union[str, yarl.URL], encoded=True):
+        self._url = yarl.URL(str(url), encoded=encoded)
+
+    def __str__(self) -> str:
+        return str(self._url)
+
+    def __repr__(self) -> str:
+        return f'{type(self).__name__}({str(self._url)!r})'
+
+    def __eq__(self, other) -> bool:
+        if self._url.path == "/":
+            if isinstance(other, str):
+                other = _Url(other)
```

Comment:
If we follow pathlib route, probably there shouldn't be magic like this:

```py
In [1]: import pathlib

In [2]: p = pathlib.Path("/etc")

In [3]: p == "/etc"
Out[3]: False
```

Reply:
You're right. Initially, I was trying to solve the tiresome casting of `str()` everytime, similar to the concern noted in https://github.com/scrapinghub/web-poet/pull/42#discussion_r888256415. One thing I realized now would be the problem it presents to end users later on. They might be expecting a `"https://example.com"` but misses out that it's actually `ResponseUrl("https://example.com")` due to this equality "magic".

Updated this on: https://github.com/scrapinghub/web-poet/pull/45/commits/912ba772bbb0d1a00955068228e515e8bcd99caf